pkg/cmd/server/handlers/impersonation_scopes.go

```diff
@@ -0,0 +1,22 @@
+package handlers
+
+import (
+	"net/http"
+
+	authenticationv1 "k8s.io/client-go/pkg/apis/authentication/v1"
+
+	authenticationapi "github.com/openshift/origin/pkg/auth/api"
+	authorizationapi "github.com/openshift/origin/pkg/authorization/apis/authorization"
+)
+
+// TranslateLegacyScopeImpersonation is a filter that will translates user scope impersonation for openshift into the equivalent kube headers.
+func TranslateLegacyScopeImpersonation(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		for _, scope := range req.Header[authenticationapi.ImpersonateUserScopeHeader] {
```

Comment:
Delete the original headers when you are done?

Reply:
> Delete the original headers when you are done?

I was going for as little mutation as possible.  I could if you really want though.

Reply:
> as little mutation as possible

Sounds reasonable enough to me.